messaging/src/test/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessorTest.java

```diff
@@ -1215,4 +1217,95 @@ private void mockSlowEventHandler() throws Exception {
             return null;
         }).when(stubEventHandler).handle(any(), any());
     }
+
+    @Test
+    void coordinatorExtendsClaimsEarlierForBusyWorkPackages() throws Exception {
+        setTestSubject(createTestSubject(builder -> builder.initialSegmentCount(1)
+                                                           .enableCoordinatorClaimExtension()));
+
+        AtomicBoolean isWaiting = new AtomicBoolean(false);
+        CountDownLatch handleLatch = new CountDownLatch(1);
+        mockEventHandlerInvoker();
+        doAnswer(invocation -> {
+            // Waiting for the latch to simulate a slow/busy WorkPackage.
+            isWaiting.set(true);
+            return handleLatch.await(5, TimeUnit.SECONDS);
+        }).when(stubEventHandler)
+          .handle(any(), any());
+
+        List<EventMessage<Integer>> events = IntStream.range(0, 42)
+                                                      .mapToObj(GenericEventMessage::new)
+                                                      .collect(Collectors.toList());
+        events.forEach(stubMessageSource::publishMessage);
+
+        testSubject.start();
+
+        // Wait until we've reached the blocking WorkPackage before validating if the token is extended.
+        // Otherwise, the WorkPackage may extend the token itself.
+        await().pollDelay(Duration.ofMillis(50))
+               .atMost(Duration.ofSeconds(5))
+               .until(isWaiting::get);
+
+        // As the WorkPackage is blocked, we can verify if the claim is extended, but not stored.
+        verify(tokenStore, timeout(5000)).extendClaim(PROCESSOR_NAME, 0);
+        verify(tokenStore, never()).storeToken(any(), eq(PROCESSOR_NAME), eq(0));
+
+        // Unblock the WorkPackage after successful validation
+        handleLatch.countDown();
+
+        // Processing finished...
+        await().pollDelay(Duration.ofMillis(50))
+               .atMost(Duration.ofSeconds(5))
+               .until(() -> testSubject.processingStatus().get(0).isCaughtUp());
+        // Validate the token is stored
+        verify(tokenStore, timeout(5000)).storeToken(any(), eq(PROCESSOR_NAME), eq(0));
+    }
+
+    @Test
+    void coordinatorExtendingClaimFailsAndAbortsWorkPackage() throws Exception {
+        setTestSubject(createTestSubject(builder -> builder.initialSegmentCount(1)
```

Comment:
you could shorted the claim extension timeout to speed up the test

Reply:
The used `createTestSubject(...)` operation already adjusts the `claimExtensionTimeout` from 5000 to 1000ms. 

However, I just tested whether 500ms is sufficient for the tests too.
Turns out it is, so the majority of tests in this class should be a little faster.